libs/ssl-config/src/main/java/org/opensearch/common/ssl/PemUtils.java

```diff
@@ -106,555 +70,88 @@ private PemUtils() {
      * @param passwordSupplier A password supplier for the potentially encrypted (password protected) key
      * @return a private key from the contents of the file
      */
-    public static PrivateKey readPrivateKey(Path keyPath, Supplier<char[]> passwordSupplier) throws IOException, GeneralSecurityException {
```

Comment:
How are we verifying all the existing cases are covered by the new implementation? I did not see any swaps of test cases to alternative formatting or handling which I would expect given the extent of these changes. Can you point me to where you are verifying that this swap to the condensed version covers all the existing patterns?

Very nice job condensing things assuming this matches up 1:1!

Reply:
The error messaging has changed, after swapping the BC libraries [PemUtilsTests.java](https://github.com/opensearch-project/OpenSearch/pull/14912/files#diff-a6c429f38b1846ea4663f6cbca818a87c0fd15b225797a5a65b53b6499934913)